src/Compilers/Shared/DesktopBuildClient.cs

```diff
@@ -473,13 +480,13 @@ private static bool CheckPipeConnectionOwnership(NamedPipeClientStream pipeStrea
             (ObjectSecurity)typeof(PipeStream)
             .GetTypeInfo()
             .GetDeclaredMethod("GetAccessControl")
-            .Invoke(pipeStream, parameters: null);
+            ?.Invoke(pipeStream, parameters: null);
 
         private static string GetUserName() =>
             (string)typeof(Environment)
             .GetTypeInfo()
             .GetDeclaredProperty("UserName")
-            .GetMethod.Invoke(null, parameters: null);
+            ?.GetMethod?.Invoke(null, parameters: null);
```

Comment:
I believe this can require certain security permissions in partial trust, but will not throw.